File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        arr = [0 for i in range(26)]

        for i in range(len(s)):
            arr[ord(s[i]) - ord('a')] += 1
            arr[ord(t[i]) - ord('a')] -= 1

        for x in arr:
            if x != 0:
                return False

        return True

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] letters = new int[26];
        int lenS = s.length();
        int lenT = t.length();
        
        if (lenS != lenT) { 
            return false;
        }
        
        for (int i = 0; i < lenS; i++) {
            letters[s.charAt(i) - 'a']++;
        }
        
        for (int i = 0; i < lenT; i++) {
            letters[t.charAt(i) - 'a']--;
            if (letters[t.charAt(i) - 'a'] < 0) {
                return false;
            } 
        }
        
        return true;
    }
}